src/sockets/chatSpamFilter.ts

```diff
@@ -0,0 +1,38 @@
+export class ChatSpamFilter {
+  history: { [username: string]: number[] };
+
+  seconds: number;
+
+  constructor() {
+    this.history = {};
+    this.seconds = 0;
+  }
+
+  // Returns true if user can chat. Else false.
+  chatRequest(username: string): boolean {
+    if (!(username in this.history)) {
+      this.history[username] = [this.seconds];
+    } else {
+      const userHistory = this.history[username];
+      userHistory.push(this.seconds);
+      if (userHistory.length > 4) {
+        if (userHistory[4] - userHistory[0] < 5) {
```

Comment:
Let's extract the 5 to a constant variable called `SPAM_WINDOW`.

Reply:
Let's also extract the 4 to a constant variable too. Maybe name it `NUM_MESSAGES`.